middleware/assert.go

```diff
@@ -39,6 +39,42 @@ func NewAssert(r responder, d datasetAPIClient, f filterFlexAPIClient, ds datast
 	}
 }
 
+// Filter Flex Forwarder that checks for filter in the route, not in the body as below.
+// Used for PUT filter-output/{filter-output-id}
+func (a *Assert) FilterOutputFilterType(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if !a.enabled {
+			next.ServeHTTP(w, r)
+			return
+		}
+
+		vars := mux.Vars(r)
+		filterOutputID := vars["filter_output_id"]
+
+		ctx := r.Context()
+		filterOutput, err := a.store.GetFilterOutput(ctx, filterOutputID)
+		if err != nil {
+			a.respond.Error(ctx, w, statusCode(err), er{
+				err: errors.Wrap(err, "failed to get filter output"),
+				msg: fmt.Sprintf("failed to get filter output"),
+			})
+			return
+		}
+
+		if filterOutput.Type == flexible {
```

Comment:
Currently I think that the `type` of `filterOutput` is not populated in the datastore. There is some work pending to correct this but for the time being it might be worth using the `filterOutput.Links.FilterBlueprint.ID` to find the `filter` with the correct `type`.

Reply:
you are right, it is actually not, having just checked. 
Although using the `filterOutput.Links.FilterBlueprint.ID` to find the filter would then result in two database calls, which feels really hacky, not sure if that is a great way to go. Would rather just say that this work is finished but blocked from releasing until the work pending to correct is finished

Reply:
@saminahbab Sounds reasonable but probably best to check with @franmoore05 and co to make sure.